qfieldsync/core/cloud_api.py

```diff
@@ -252,22 +373,33 @@ def server_url(self):
 
         return re.sub(r"([^:]/)(/)+", r"\1", url)
 
+    def get_auth_capabilities(self) -> QNetworkReply:
+        """Get authentication capabilities from a server"""
+        reply = self.cloud_get("auth/providers/")
+        return reply
+
     def auto_login_attempt(self):
-        cfg = self.auth()
+        if self.auth_method == CloudAuthMethod.CREDENTIALS:
+            cfg = self.auth()
 
-        server_url = cfg.uri() or self.url
-        username = cfg.config("username")
-        password = cfg.config("password")
+            server_url = cfg.uri() or self.url
+            username = cfg.config("username")
+            password = cfg.config("password")
 
-        if username and password:
-            self.set_url(server_url)
-            self.login(username, password)
+            if all([username, password]):
+                self.set_url(server_url)
+            self.login_with_credentials(username, password)
 
-    def login(self, username: str, password: str) -> Optional[QNetworkReply]:
-        """Login to QFieldCloud"""
+        elif self.auth_method == CloudAuthMethod.SSO:
+            self.login_with_sso()
+
+    def login_with_credentials(
+        self, username: str, password: str
+    ) -> Optional[QNetworkReply]:
+        """Login to QFieldCloud with classic login/password method"""
```

Comment:
```suggestion
        """Login to QFieldCloud with login/password method"""
```
Just to prevent someone to confuse it with QFieldCloud classic.